### DISCUSSION TITLE
Can't run properly under authelia Docker?  （ I just want to test it out?）

### DISCUSSION THREAD

**Discussion by angel20009123** [authelia/authelia#8270] [Q&A] [answered]
docker run -d --name authelia -p 9091:9091 \
  -v /root/authelia/configuration.yml:/config/configuration.yml \
  -v /root/authelia/users_database.yml:/config/users_database.yml \
  authelia/authelia:latest

[root@Rocky9 config]# docker images
REPOSITORY          TAG       IMAGE ID       CREATED        SIZE
authelia/authelia   4.38.17   dd3a6c501ab2   2 weeks ago    55.5MB
authelia/authelia   latest    dd3a6c501ab2   2 weeks ago    55.5MB
nginx               latest    60c8a892f36f   6 weeks ago    192MB
alpine              latest    63b790fccc90   2 months ago   7.8MB
[root@Rocky9 config]# docker ps -a
CONTAINER ID   IMAGE                      COMMAND                CREATED         STATUS                     PORTS     NAMES
f26784ad3a17   authelia/authelia:latest   "/app/entrypoint.sh"   3 minutes ago   Exited (1) 3 minutes ago             authelia
[root@Rocky9 config]# 

[root@Rocky9 config]# cat configuration.yml 
server:
  address: "tcp://0.0.0.0:9091"
  jwt_secret: "mysecretkey123"

authentication_backend:
  file:
    path: "/config/users_database.yml"

session:
  domain: "example.com"
  cookies:
    http_only: true
    same_site: strict
    secure: true

access_control:
  default_policy: two_factor
  rules:
    - domain: "example.com"
      policy: two_factor
      groups:
        - "admin"

storage:
  local:
    path: "/config/authelia_storage"

log:
  level: debug
  file:
    path: "/config/logs/access.log"

notifier:
  smtp:
    host: "smtp.example.com"
    port: 587
    username: "user@example.com"
    password: "password"
    from: "user@example.com"
    to:
      - "admin@example.com"

identity_validation:
  reset_password:
    jwt_secret: "your_jwt_secret"

[root@Rocky9 config]# 

[root@Rocky9 config]# cat users_database.yml 
users:
  admin:
    password: "adminpassword" 
    groups:
      - "admin"
  user1:
    password: "userpassword"
    groups:
      - "user"

[root@Rocky9 config]#

**Comment by james-d-elliott** [comment]
Authelia wont start with an invalid configuration. You have several errors, try the following adjustments:

```diff
3d2
<   jwt_secret: "mysecretkey123"
10d8
<   domain: "example.com"
12,14c10,11
<     http_only: true
<     same_site: strict
<     secure: true
---
>     - domain: "example.com"
>       authelia_url: "https://auth.example.com"
21,22c18,19
<       groups:
<         - "admin"
---
>       subject:
>        - 'group:admin'
24a22
>   encryption_key: "you_must_include_an_encryption_key"
30,31d27
<   file:
<     path: "/config/logs/access.log"
35,36c31
<     host: "smtp.example.com"
<     port: 587
---
>     address: "submission://smtp.example.com:587"
39,41c34
<     from: "user@example.com"
<     to:
<       - "admin@example.com"
---
>     sender: "user@example.com"
```

**Reply by angel20009123** [reply]
I can understand it as running to detect issues with the configuration running results. Will Docker fail to start? (If I have an LDAP server with the correct configuration, but there is a firewall in front of LDAP that allows me to access the IP. Will Docker also fail to code?)

**Reply by james-d-elliott** [reply]
Docker doesn't fail to start exactly, authelia logs the errors and exits on an invalid config. If you look at the docker logs it will show this. We also ensure the settings for each service like LDAP, SMTP, etc are valid before starting.

**Reply by angel20009123** [reply]
I set up a container to run, and the configuration is also new. Why did you package this error?, My configuration above does not have any LDAP related information





[root@Rocky9 config]# chmod +x run_authelia.sh
[root@Rocky9 config]# ./run_authelia.sh
b5817747b51e317f6d6e62a23f3a3ca0540bab7ae0a237e75d2acfe26c7cc250
Authelia container started successfully.
[root@Rocky9 config]# docker ps -a
CONTAINER ID   IMAGE                      COMMAND                CREATED         STATUS                     PORTS     NAMES
b5817747b51e   authelia/authelia:latest   "/app/entrypoint.sh"   9 seconds ago   Exited (1) 8 seconds ago             authelia
[root@Rocky9 config]# docker logs authelia
time="2024-11-15T09:02:22Z" level=warning msg="Configuration: configuration key 'authentication_backend.ldap.url' is deprecated in 4.38.0 and has been replaced by 'authentication_backend.ldap.address': you are not required to make any changes as this has been automatically mapped for you, but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-11-15T09:02:22Z" level=warning msg="Configuration: configuration keys 'server.host', 'server.port', and 'server.path' are deprecated in 4.38.0 and has been replaced by 'server.address' in the format of '[tcp[(4|6)]://]<hostname>[:<port>][/<path>]' or 'tcp[(4|6)://][hostname]:<port>[/<path>]': you are not required to make any changes as this has been automatically mapped for you to the value 'tcp://0.0.0.0:9091/', but to stop this warning being logged you will need to adjust your configuration, and this configuration key and auto-mapping is likely to be removed in 5.0.0"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: error occurred during unmarshalling configuration: decoding failed due to the following error(s):\n\n'authentication_backend.ldap.address' expected a map, got 'slice'\n'authentication_backend.ldap.user' expected type 'string', got unconvertible type 'map[string]interface {}', value: 'map[base_dn:OU=Big Data,DC=dotcomlab,DC=net display_name_attribute:CN dn:CN=joinad,OU=Big Data,DC=dotcomlab,DC=net filter:(sAMAccountName={0}) mail_attribute:mail password:K#&wym4loUYM username_attribute:sAMAccountName]'"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.group.base_dn"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.group.filter"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.group.name_attribute"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.group.user_attribute"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.base_dn"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.display_name_attribute"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.dn"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.filter"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.mail_attribute"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.password"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: authentication_backend.ldap.user.username_attribute"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: log.file.max_age"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: log.file.max_backups"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: log.file.max_size"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: log.file.path"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: second_factor.otp.digits"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: second_factor.otp.issuer"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: second_factor.otp.period"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: server.jwt_secret"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: session.cookies.http_only"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: session.cookies.same_site"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: session.secure"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: configuration key not expected: access_control.rules[].groups"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: authentication_backend: you must ensure either the 'file' or 'ldap' authentication backend is configured"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: session: option 'domain' and option 'cookies' can't be specified at the same time"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: session: domain config #1 (domain ''): option 'domain' is required"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: notifier: you must ensure either the 'smtp' or 'filesystem' notifier is configured"
time="2024-11-15T09:02:22Z" level=error msg="Configuration: identity_validation: reset_password: option 'jwt_secret' is required when the reset password functionality isn't disabled"
time="2024-11-15T09:02:22Z" level=fatal msg="Can't continue due to the errors loading the configuration" stack="github.com/authelia/authelia/v4/internal/commands/context.go:327 (*CmdCtx).ConfigValidateLogRunE\ngithub.com/authelia/authelia/v4/internal/commands/context.go:186 NewRootCmd.(*CmdCtx).ChainRunE.func1\ngithub.com/spf13/cobra@v1.8.1/command.go:970                     (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                    (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                    (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10          main\ninternal/runtime/atomic/types.go:194                             (*Uint32).Load\nruntime/asm_amd64.s:1700                                         goexit"
[root@Rocky9 config]# ll
total 12
-rw-r--r--. 1 root root 811 Nov 15 16:55 configuration.yml
-rwxr-xr-x. 1 root root 292 Nov 15 17:01 run_authelia.sh
-rw-r--r--. 1 root root 220 Nov 15 15:08 users_database.yml
[root@Rocky9 config]#

**Reply by angel20009123** [reply]
[root@Rocky9 config]# cat configuration.yml 
server:
  address: "tcp://0.0.0.0:9091"
  jwt_secret: "mysecretkey123"

authentication_backend:
  file:
    path: "/config/users_database.yml"

session:
  domain: "authelia.kq.gov.cn"
  cookies:
    http_only: true
    same_site: strict
    secure: true

access_control:
  default_policy: two_factor
  rules:
    - domain: "authelia.kq.go.cn"
      policy: two_factor
      groups:
        - "admin"

storage:
  local:
    path: "/config/authelia_storage"

log:
  level: debug
  file:
    path: "/config/logs/access.log"

notifier:
  smtp:
    host: "10.59.0.3"
    port: 587
    username: "authelia1@dotcomlab.net"
    password: "1qaz@WSX3edc"
    from: "authelia1@dotcomlab.net"
    to:
      - "huanggq@dotcomlab.net"

identity_validation:
  reset_password:
    jwt_secret: "NPxAQnEbahbRksv4YjdLRI9ws2mpfu"

[root@Rocky9 config]#

**Reply by james-d-elliott** [reply]
> I set up a container to run, and the configuration is also new. Why did you package this error?, My configuration above does not have any LDAP related information

The errors exist because you do have it configured. Show the output of the `cat /root/authelia/configuration.yml` cmd, not `cat configuration.yml` from the `config` dir.

**Reply by angel20009123** [reply]
I want GPT to verify and modify this configuration file. Always wrong?

"/config/configuration.yml" 43L, 817B written
[root@Rocky9 ~]# authelia-linux-amd64 validate-config --config /config/configuration.yml
Configuration parsed and loaded with errors:

         - configuration key not expected: notifier.smtp.recipients
         - configuration key not expected: session.domains
         - session: option 'cookies' is required

[root@Rocky9 ~]# vi /config/configuration.yml 
identity_validation:
  reset_password:
    jwt_secret: "averylongjwtsecretkey123"

server:
  address: "tcp://0.0.0.0:9091"

authentication_backend:
  file:
    path: "/config/users_database.yml"

session:
  domains:
    - "authelia.kq.gov.cn"
  same_site: strict

access_control:
  default_policy: two_factor
  rules:
    - domain: "authelia.kq.gov.cn"
      policy: two_factor
      subject:
        - "group:admin"

storage:
  encryption_key: "averylongrandomencryptionkey123"
  local:
    path: "/config/authelia_storage"

log:
  level: debug
  format: text
  file_path: "/config/logs/access.log"

notifier:
  smtp:
    address: "submission://10.59.0.3:587"
    username: "authelia1@dotcomlab.net"
    password: "1qaz@WSX3edc"
    sender: "authelia1@dotcomlab.net"
    recipients:
      - "huanggq@dotcomlab.net"
identity_validation:
  reset_password:
    jwt_secret: "averylongjwtsecretkey123"

server:
  address: "tcp://0.0.0.0:9091"

authentication_backend:
  file:
    path: "/config/users_database.yml"

session:
  domain: "authelia.kq.gov.cn"
  cookies:
    http_only: true
    same_site: strict
    secure: true

access_control:
  default_policy: two_factor
  rules:
    - domain: "authelia.kq.gov.cn"
      policy: two_factor
      subject:
        - "group:admin"

storage:
  encryption_key: "averylongrandomencryptionkey123"
  local:
    path: "/config/authelia_storage"

log:
  level: debug
  format: text
  file_path: "/config/logs/access.log"

notifier:
  smtp:
    address: "submission://10.59.0.3:587"
    username: "authelia1@dotcomlab.net"
    password: "1qaz@WSX3edc"
    sender: "authelia1@dotcomlab.net"
    to:
      - "huanggq@dotcomlab.net"

"/config/configuration.yml" 45L, 852B written
[root@Rocky9 ~]# authelia-linux-amd64 validate-config --config /config/configuration.yml
Configuration parsed and loaded with errors:

         - configuration key not expected: notifier.smtp.to
         - configuration key not expected: session.cookies.http_only
         - configuration key not expected: session.cookies.same_site
         - configuration key not expected: session.cookies.secure
         - session: option 'domain' and option 'cookies' can't be specified at the same time
         - session: domain config #1 (domain ''): option 'domain' is required

[root@Rocky9 ~]#

**Comment by james-d-elliott** [comment]
I'm not sure why you're trying to create a GPT for this but we provide configuration samples along with all of the docs related to each option and that's why the validator exists. 

Configuration examples of the areas you got wrong:

session: https://www.authelia.com/configuration/session/introduction/#configuration
smtp: https://www.authelia.com/configuration/notifications/smtp/#configuration

Also both of these sections I showed you corrections in this diff: https://github.com/authelia/authelia/discussions/8270#discussioncomment-11264207

**Reply by angel20009123** [reply]
I have passed the validator, but there is still a problem with starting and an error message. I am very depressed



[root@Rocky9 config]# authelia-linux-amd64 validate-config --config /config/configuration.yml
Configuration parsed and loaded successfully without errors.

[root@Rocky9 config]# cat users_database.yml 
users:
  admin:
    password: "$argon2id$v=19$m=65536,t=3,p=4$7cy9knEzF/MJOnHLyywjhw$upJ5zOk3HAulolmXumcpdvwU3tazrt7T+SUzZsITcZE"
    email: authelia1@dotcomlab.net
    groups:
      - "admin"

[root@Rocky9 config]# ll
total 12
drwxr-xr-x. 2 root root    6 Nov 16 10:05 authelia_storage
-rw-r--r--. 1 root root 1907 Nov 16 13:07 configuration.yml
drwxr-xr-x. 2 root root   24 Nov 16 11:52 logs
-rwxr-xr-x. 1 root root  279 Nov 16 09:56 run_authelia.sh
-rw-r--r--. 1 root root  194 Nov 16 13:30 users_database.yml
[root@Rocky9 config]# ./run_authelia.sh 
8d5f4193688ab9298836fee4226b3882e5f89de15eccb237ebdc71b9c0f95f07
Authelia container started successfully.
[root@Rocky9 config]# docker ps -a
CONTAINER ID   IMAGE                      COMMAND                CREATED         STATUS                     PORTS     NAMES
8d5f4193688a   authelia/authelia:latest   "/app/entrypoint.sh"   4 seconds ago   Exited (1) 2 seconds ago             authelia
[root@Rocky9 config]# docker logs authelia
time="2024-11-16T05:30:46Z" level=debug msg="Loaded Configuration Sources" files="[/config/configuration.yml]" filters="[]"
time="2024-11-16T05:30:46Z" level=debug msg="Logging Initialized" fields.level=debug file= format=text keep_stdout=true
time="2024-11-16T05:30:46Z" level=debug msg="Process user information" gid=0 gids="1,2,3,4,6,10,11,20,26,27" name=root uid=0 username=root
time="2024-11-16T05:30:46Z" level=info msg="Authelia v4.38.17 is starting"
time="2024-11-16T05:30:46Z" level=info msg="Log severity set to debug"
time="2024-11-16T05:30:46Z" level=info msg="Storage schema is being checked for updates"
time="2024-11-16T05:30:46Z" level=info msg="Storage schema migration from 0 to 15 is being attempted"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 0 to 1"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 1 to 2"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 2 to 3"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 3 to 4"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 4 to 5"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 5 to 6"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 6 to 7"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 7 to 8"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 8 to 9"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 9 to 10"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 10 to 11"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 11 to 12"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 12 to 13"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 13 to 14"
time="2024-11-16T05:30:46Z" level=debug msg="Storage schema migrated from version 14 to 15"
time="2024-11-16T05:30:46Z" level=info msg="Storage schema migration from 0 to 15 is complete"
time="2024-11-16T05:30:46Z" level=error msg="Error occurred running a startup check" error="error reading the authentication database: could not validate the schema: Users.admin.users: non zero value required" provider=user stack="github.com/authelia/authelia/v4/internal/commands/root.go:118 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:88  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.1/command.go:985                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\ninternal/runtime/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1700                                      goexit"
time="2024-11-16T05:30:46Z" level=fatal msg="One or more providers had fatal failures performing startup checks, for more detail check the error level logs" providers="[user]" stack="github.com/authelia/authelia/v4/internal/commands/root.go:150 doStartupChecks\ngithub.com/authelia/authelia/v4/internal/commands/root.go:88  (*CmdCtx).RootRunE\ngithub.com/spf13/cobra@v1.8.1/command.go:985                  (*Command).execute\ngithub.com/spf13/cobra@v1.8.1/command.go:1117                 (*Command).ExecuteC\ngithub.com/spf13/cobra@v1.8.1/command.go:1041                 (*Command).Execute\ngithub.com/authelia/authelia/v4/cmd/authelia/main.go:10       main\ninternal/runtime/atomic/types.go:194                          (*Uint32).Load\nruntime/asm_amd64.s:1700                                      goexit"
[root@Rocky9 config]# ll
total 12
drwxr-xr-x. 2 root root    6 Nov 16 10:05 authelia_storage
-rw-r--r--. 1 root root 1907 Nov 16 13:07 configuration.yml
drwxr-xr-x. 2 root root   24 Nov 16 11:52 logs
-rwxr-xr-x. 1 root root  279 Nov 16 09:56 run_authelia.sh
-rw-r--r--. 1 root root  194 Nov 16 13:30 users_database.yml
[root@Rocky9 config]# cat configuration.yml 
identity_validation:
  reset_password:
    jwt_secret: "averylongjwtsecretkey123"

server:
  address: "tcp://0.0.0.0:9091"

authentication_backend:
  file:
    path: "/config/users_database.yml"

session:
  secret: 'averysecurelongsessionsecret'  # 
  name: 'authelia_session'
  same_site: 'lax'
  inactivity: '5m'
  expiration: '1h'
  remember_me: '1M'
  cookies:
    - domain: 'kq.gov.cn'
      authelia_url: 'https://authelia.kq.gov.cn'
      default_redirection_url: 'https://dns120.kq.gov.cn:10443/#/'
      name: 'authelia_session'
      same_site: 'lax'
      inactivity: '5m'
      expiration: '1h'
      remember_me: '1d'

access_control:
  default_policy: two_factor
  rules:
    - domain: "authelia.kq.gov.cn"
      policy: two_factor
      subject:
        - "group:admin"

storage:
  encryption_key: "averylongrandomencryptionkey123"
  local:
    path: "/config/authelia_storage"

log:
  level: debug
  format: text
  keep_stdout: true

notifier:
  disable_startup_check: false
  smtp:
    address: 'smtp://10.59.0.3:587'  # 
    timeout: '5s'
    username: 'authelia1@dotcomlab.net'  # 
    password: '1qaz@WSX3edc'  # 
    sender: "Authelia <authelia1@dotcomlab.net>"  # 
    identifier: 'authelia.kq.gov.cn'  # 
    subject: "[Authelia] {title}"  # 
    startup_check_address: 'huanggq@dotcomlab.net'  # 
    disable_require_tls: false
    disable_starttls: false
    disable_html_emails: false
    tls:
      server_name: '10.59.0.3'  # 
      skip_verify: true
      minimum_version: 'TLS1.2'
      maximum_version: 'TLS1.3'

ntp:
  address: 'udp://10.75.162.10:123'
  version: 3
  max_desync: '3s'
  disable_startup_check: false
  disable_failure: false
[root@Rocky9 config]#

**Reply by james-d-elliott** [reply]
Last you showed `/root/authelia/users_database.yml` was the file mounted, not a file in `config/users_database.yml`. Show the docker run/docker compose being used.

**Reply by angel20009123** [reply]
I made a mistake about which directory last time, I am currently running this one

[root@Rocky9 ~]# cd /config/
[root@Rocky9 config]# ll
total 12
drwxr-xr-x. 2 root root    6 Nov 16 10:05 authelia_storage
-rw-r--r--. 1 root root 1907 Nov 16 13:07 configuration.yml
drwxr-xr-x. 2 root root   24 Nov 16 11:52 logs
-rwxr-xr-x. 1 root root  279 Nov 16 09:56 run_authelia.sh
-rw-r--r--. 1 root root  194 Nov 16 13:30 users_database.yml
[root@Rocky9 config]# cat run_authelia.sh 
#!/bin/bash

# Run the Authelia container
docker run -d --name authelia -p 9091:9091 \
-v /config/configuration.yml:/config/configuration.yml \
-v /config//users_database.yml:/config/users_database.yml \
authelia/authelia:latest

echo "Authelia container started successfully."

[root@Rocky9 config]#

**Reply by james-d-elliott** [reply]
Not sure why the error is so obscure but the reason is lack of the `displayname` value which is required for every user.

**Reply by angel20009123** [reply]
Do we need to define it like this in the YAML file of the database
users:
  admin:
    password: $argon2i$v=19$m=1024,t=2,p=8$hash_here  # 替换为实际的哈希密码
    email: admin@example.com
    displayname: "Admin User" 
    groups:
      - admin
  user1:
    password: $argon2i$v=19$m=1024,t=2,p=8$hash_here  # 替换为实际的哈希密码
    email: user1@example.com
    displayname: "Regular User 1"
    groups:
      - user

**Reply by angel20009123** [reply]
Docker is up now

[root@Rocky9 config]# docker ps -a
CONTAINER ID   IMAGE                      COMMAND                CREATED          STATUS                    PORTS                                       NAMES
8b39db2ee175   authelia/authelia:latest   "/app/entrypoint.sh"   37 minutes ago   Up 37 minutes (healthy)   0.0.0.0:9091->9091/tcp, :::9091->9091/tcp   authelia
[root@Rocky9 config]# 
![image](https://github.com/user-attachments/assets/1678bba9-5dce-44e8-b250-07788c99cf43)
？？？？？？？

**Reply by james-d-elliott** [reply, marked as answer]
`docker logs authelia`
